include/boost/wave/util/cpp_iterator.hpp

```diff
@@ -2456,7 +2455,6 @@ pp_iterator_functor<ContextT>::on_pragma(
 #endif
         }
     }
-    expanded.push_back(result_type(T_NEWLINE, "\n", act_token.get_position()));
```

Comment:
Removing the final new line will cause emitted pragmas to miss the final new line and be misintepreted by following compile steps. For example
```
#pragma foo
#define X hello
X
```
will currently get preprocessed to
```
#pragma foo#define X hello
X
```
but should be actually
```
#pragma foo
hello
```
Shall i add an issue with more details?